### PR TITLE
[FLOC-2538] An option to allow all flocker scripts to log to a file

### PR DESCRIPTION
**Pull request by wallrj**
Design for https://clusterhq.atlassian.net/browse/FLOC-2538

### CODE REVIEW

File: admin/package-files/upstart/flocker-container-agent.conf [hunk 1]
```diff
@@ -9,12 +9,12 @@ stop on runlevel [016]
 respawn
 
 pre-start script
-	if [ ! -r /etc/flocker/agent.yml ]; then
-		echo "Cannot read configuration file '/etc/flocker/agent.yml'."
-		exit 1
-	fi
+        if [ ! -r /etc/flocker/agent.yml ]; then
```
Comment (itamarst) [inline admin/package-files/upstart/flocker-container-agent.conf]: Changing from tabs to spaces might break this, make sure that's OK.

File: flocker/common/script.py [hunk 2]
```diff
@@ -74,6 +76,10 @@ def main(reactor, options):
 
 
 def eliot_logging_service(log_file, reactor, capture_stdout):
+    # FLOC-2538 - So log_file may be an instance of
```
Comment (itamarst) [inline flocker/common/script.py]: Should be fine, `LogFile` is sufficiently similar to a normal `file` that  ThreadedFileWriter will be happy.

**Comment by itamarst** [comment]
The branch name is wrong, I think you want `release-maintenance/flocker-1.0.0pre2/...` (i.e. match current thing you want to merge into?). Also we'll hopefully have `release/flocker-1.0.2` soon. And this scheme is soooo terrible and error prone. Hopefully we can fix it for next release.

**Comment by itamarst** [comment]
Anyway looks good please proceed.

**Comment by wallrj** [comment]
I got the branch name wrong. So I'll close this PR and the branch.

**Pull request closed without merging**